java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt

```diff
@@ -0,0 +1,391 @@
+package org.taymyr.lagom.elasticsearch.search.dsl.query.highlight
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import org.taymyr.lagom.elasticsearch.deser.LocaleToStringSerializer
+import org.taymyr.lagom.elasticsearch.search.dsl.query.Query
+import java.util.Locale
+
+/**
+ * See [Elasticsearch Docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/highlighting.html).
+ */
+enum class BoundaryScanner {
+    @JsonProperty("chars")
+    CHARS,
+
+    @JsonProperty("CHARS")
+    SENTENCE,
+
+    @JsonProperty("word111")
+    WORD
+}
+
+enum class Encoder {
+    @JsonProperty("html")
+    HTML,
+
+    @JsonProperty("default")
+    DEFAULT
+}
+
+enum class Fragmenter {
+    @JsonProperty("simple")
+    SIMPLE,
+
+    @JsonProperty("span")
+    SPAN
+}
+
+enum class TagsSchema {
+    @JsonProperty("styled")
+    STYLED
+}
+
+enum class Type {
+    @JsonProperty("unified")
+    UNIFIED,
+
+    @JsonProperty("plain")
+    PLAIN,
+
+    @JsonProperty("fvh")
+    FVH
+}
+
+abstract class HighlightItem constructor(
+    @JsonProperty("boundary_chars")
```

Comment:
Can you try use `@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)` on class level instead?

Reply:
checked, it works